libOPHD/Population/PopulationPool.h

```diff
@@ -15,16 +15,16 @@ class PopulationPool
 public:
 	void population(Population* pop);
 
-	int availableWorkers();
-	int availableScientists();
+	const int availableWorkers() const;
+	const int availableScientists() const;
 
 	bool usePopulation(PopulationRequirements populationRequirements);
 
 	void clear();
 
 	int scientistsAsWorkers();
-	int scientistsEmployed();
-	int workersEmployed();
+	const int scientistsEmployed() const;
+	const int workersEmployed() const;
 	int populationEmployed();
```

Comment:
scientistsAsWorkers, populationEmployed, and size have a similar behavior as the other functions. Does it make sense to change them too. I made the changes locally and it did not appear to create any issues.

Reply:
It's often a good idea to marking functions as `const` as a sort of default, and only remove it if the function really needs to mutate state.

A quick note on semantics: I've seen cases where you could technically mark a function as `const`, and it would compile, but that was only because there was a level of indirection to the data being modified. The internal level of indirection is an implementation detail, and the nature of the function in that case really was to mutate state. In such cases, I think it makes the most sense to not declare a method as `const`. In particular, if you were to refactor the internal implementation, you might be required to remove the `const`, which could potentially break calling code if it assumed a method was callable on a `const` object.

Reply:
It doesn't really make sense to declare the return values as `const`. They are being copied anyway.

Besides, the caller should decide if the result they're storing should be `const` or not.